cnd/src/db/tables.rs

```diff
@@ -0,0 +1,733 @@
+use crate::{
+    db::{
+        schema::{
+            address_hints, halight_swap_details, herc20_swap_details, secret_hashes,
+            shared_swap_ids, swaps,
+        },
+        wrapper_types::{
+            custom_sql_types::{Text, U32},
+            BitcoinNetwork, Erc20Amount, EthereumAddress, Satoshis,
+        },
+        Error, Sqlite,
+    },
+    lightning,
+    swap_protocols::{self, rfc003, HashFunction, LocalSwapId, Role},
+};
+use diesel::{self, prelude::*, RunQueryDsl};
+use libp2p::{Multiaddr, PeerId};
+
+// TODO: Audit 'pub' scope on all structs.
+
+#[derive(Identifiable, Queryable, PartialEq, Debug)]
+#[table_name = "swaps"]
+pub struct Swap {
+    id: i32,
+    local_swap_id: Text<LocalSwapId>,
+    role: Text<Role>,
+    counterparty_peer_id: Text<PeerId>,
+}
+
+impl From<Swap> for ISwap {
+    fn from(swap: Swap) -> Self {
+        ISwap {
+            local_swap_id: swap.local_swap_id,
+            role: swap.role,
+            counterparty_peer_id: swap.counterparty_peer_id,
+        }
+    }
+}
+
+#[derive(Insertable, Debug, Clone)]
+#[table_name = "swaps"]
+pub struct ISwap {
```

Comment:
I know it is longer but I was confused about the `I` because it means `Interface` in other languages.
```suggestion
pub struct InsertableSwap {
```

Reply:
I knew review would say this :(  I disagree but will change it at 99% review if requested.  For now I'm going to leave it like this because its me typing `Insertable` 400 times a day. Its a few simple perl one-liners to change later.

Reply:
Can you please set up auto-completion in your editor so you don't have to type these things manually and we can stop these discussions?

For the record, if I type `ISwap` and hit tab, it will autocomplete all names that are `IwhatevertextishereSwap` because it recognizes capitals. Surely there is something for emacs that can do that.

Reply:
I have auto-completion set up, I still don't like it :)

Reply:
FTR I tried this, we get naming clashes with rfc003 `Swap` and `InsertableSwap` - I'm going to leave it as `ISwap` until the rfc003 stuff is gone out of the db module.